x/crosschain/keeper/msg_server_vote_inbound_tx.go

```diff
@@ -97,23 +97,14 @@ func (k msgServer) VoteInbound(
 	if !finalized {
 		return &types.MsgVoteInboundResponse{}, nil
 	}
-	tss, tssFound := k.zetaObserverKeeper.GetTSS(ctx)
-	if !tssFound {
-		return nil, types.ErrCannotFindTSSKeys
-	}
-	// create a new CCTX from the inbound message.The status of the new CCTX is set to PendingInbound.
-	cctx, err := types.NewCCTX(ctx, *msg, tss.TssPubkey)
-	if err != nil {
-		return nil, err
-	}
-	// Initiate outbound, the process function manages the state commit and cctx status change.
-	// If the process fails, the changes to the evm state are rolled back.
-	_, err = k.InitiateOutbound(ctx, &cctx)
+
+	cctx, err := k.ValidateInboundObservers(ctx, msg, true)
 	if err != nil {
 		return nil, err
 	}
+
 	// Save the inbound CCTX to the store. This is called irrespective of the status of the CCTX or the outcome of the process function.
-	k.SaveInbound(ctx, &cctx, msg.EventIndex)
+	k.SaveInbound(ctx, cctx, msg.EventIndex)
```

Comment:
I think this method could be renamed `SaveObservedInboundInformation` to make explicity it is related to observer gateway.

I also think it would be proper to put this one above to have `ValidateInbound` as the final call. The data saved here are independent of the outbound logic